### PR TITLE
Remove unused linker inputs "-Wunused-command-line-argument"

### PR DESCRIPTION
**Pull request by jkottiku**
SWDEV-303680
[ROCmValidationSuite] [Compiler Warning] Unused linker input "-Wunused-command-line-argument

Remove unused linker command line inputs from compiler flags (C FLAGS).

### CODE REVIEW

**Review by frepaul — COMMENTED** [review]
We dont have HCC anymore. may be it is good to have a name used correctly as HIPCC_CXX_FLAGS?

**Comment by jkottiku** [comment]
> We dont have HCC anymore. may be it is good to have a name used correctly as HIPCC_CXX_FLAGS?

Sure. Planning to create a ticket to bring all the common compiler variables to the main CMakeLists.txt. Will rename HCC_CXX_FLAGS to HIPCC_CXX_FLAGS there.

File: CMakeLists.txt [hunk 1]
```diff
@@ -431,7 +431,7 @@ if(BUILD_ADDRESS_SANITIZER)
   set(ASAN_LD_FLAGS "-fuse-ld=lld")
 endif()
 
-set(HCC_CXX_FLAGS  "-Xlinker --enable-new-dtags -fno-gpu-rdc  --amdgpu-target=gfx90a --amdgpu-target=gfx1030 --amdgpu-target=gfx803 --amdgpu-target=gfx900 --amdgpu-target=gfx906 --amdgpu-target=gfx908")
```
Comment (manoj-freyr) [inline CMakeLists.txt:434]: hope this doesnt have implications to rpath related issues

**Review by manoj-freyr — APPROVED** [review]
LGTM.

File: CMakeLists.txt (hunk 1 above)
Comment (jkottiku) [reply CMakeLists.txt:434]: No, as we were including these options in C FLAGS rather than LD FLAGS.

**Pull request merged**